### DISCUSSION TITLE
GLTF to NGT components

### DISCUSSION THREAD

**Discussion by IRobot1** [nartc/angular-three#108] [Ideas]
While exploring react-three-filber, I noticed the [GLTF JSX](https://github.com/pmndrs/gltfjsx) tool.

I was thinking that an interesting web app could be developed that allowed loading a GLTF (or other model format) file and generating NGT components.

The scene graph could be viewed in a tree.  The user could pick which nodes would map to an NGT component for animating or physics.   The app would then allows exporting those nodes to NGT components (html/ts) using ComponentStore and physics best practices.

Modelling by hand using basic geometry in NGT isn't easy and mapping loaded models to NGT types manually isn't great either.

Any thoughts?  I suppose a quick prototype would help flush out the idea.  Could be a big time saver and make NGT more attractive to use.

**Comment by nartc** [comment]
Yeah, I have the same idea. Maybe a port of GLTF JSX would be fine.

**Reply by nartc** [reply]
Looks like the top level component ( in the generated JSX ) is always a `group` which is something we can do for sure.
